server/fishtest/templates/tests_view.mak

```diff
@@ -559,22 +566,23 @@
       return showDiff(diffId);
     }
 
-    function fetchDiff() {
-      return fetch(diffApiUrl, {
+    async function fetchDiff() {
+      const response = await fetch(diffApiUrl, {
         headers: {
           Accept: "application/vnd.github.v3.diff",
         },
-      })
-        .then((response) => {
-          if (response.ok) return response.text();
-        })
-        .then((text) => {
-          if (!text) return Promise.resolve();
-          return showDiff(text);
-        });
+      });
+      if(response.ok) {
+        var text = await response.text();
```

Comment:
The deceleration of this variable here will most certainly make it undefined if the response was not ok, resulting in a fatal runtime error, the same goes for the json variable.

better is to declare the variable with a null value outside before the if checking.

Reply:
This is just a literal translation of the current code (which has no error handling) to async/await. I would like to keep this as a separate PR. Later we can clean up the code and add error handling, which is easy in the async/await framework. The reason for not doing it now is that it is laborious to test the error handling code paths, and moreover error handling may paper over genuine bugs.